src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -0,0 +1,293 @@
+                                                                   args.C,
+                                                                   args.input,
+                                                                   args.filter,
+                                                                   args.input,
+                                                                   args.strides,
+                                                                   args.dilation,
+                                                                   args.lPadding,
+                                                                   args.rPadding);
+    return conv_ptrs[this->index].IsSupportedArgument(argument_ptr.get());
+#endif
+}
+
+bool PerformanceConfigHipImplicitGemmFwdXdlops::operator==(
+    const PerformanceConfigHipImplicitGemmFwdXdlops& other) const
+{
+    return this->index == other.index;
+}
+
+PerformanceConfigHipImplicitGemmFwdXdlops
+ConvHipImplicitGemmFwdXdlops::GetDefaultPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    PerformanceConfigHipImplicitGemmFwdXdlops pp;
+    pp.HeuristicInit(ctx);
+    return pp;
+}
+
+bool ConvHipImplicitGemmFwdXdlops::IsValidPerformanceConfig(
+    const ConvolutionContext& ctx, const PerformanceConfigHipImplicitGemmFwdXdlops& config) const
+{
+    return config.IsValid(ctx);
+}
+
+PerformanceConfigHipImplicitGemmFwdXdlops
+ConvHipImplicitGemmFwdXdlops::Search(const ConvolutionContext& ctx,
+                                     const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, invoke_ctx);
+}
+
+size_t ConvHipImplicitGemmFwdXdlops::GetWorkspaceSize(const ConvolutionContext& ctx) const
+{
+    std::ignore = ctx;
+    return 0;
+}
+
+bool ConvHipImplicitGemmFwdXdlops::IsApplicable(const ConvolutionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(!ctx.use_hip_kernels)
+        return false;
+    if(!(ctx.conv_problem.GetInDataType() == miopenInt8 &&
+         ctx.conv_problem.GetWeightsDataType() == miopenInt8 &&
+         ctx.conv_problem.GetOutDataType() == miopenInt8))
+        return false;
+    if(!ctx.direction.IsForward())
+        return false;
+    if(!ctx.Is2d())
+        return false;
+    if(ctx.GetStream().GetDeviceName() != "gfx908")
+        return false;
+    if(!ctx.IsLayoutNHWC())
+        return false;
+
+    std::vector<DeviceConvFwdPtr_t> conv_ptrs;
+    add_device_conv2d_fwd_xdl_nhwc_kyxc_nhwk_int8_instances_t(conv_ptrs);
+    assert(!conv_ptrs.empty());
+    const auto args = CKArgs{ctx};
+    if(!std::all_of(args.strides.begin(), args.strides.end(), [&](auto x) { return x == 1; }))
+        return false;
+
+    for(auto& conv_ptr : conv_ptrs)
```

Comment:
[Q] I am curious how fast is this loop. Slow IsApplicable may lead to global performance degradation.

Reply:
I have not benchmarked this, however, my understanding is that the trip count for this loop should not exceed a 100 ( if not less) and should not have a considerable impact on performance.

Reply:
@JehandadKhan Yeah, but I do not know what is how DeviceConvFwdPtr_t::MakeArgumentPointer() is implemented (in CK library, right?), how fast is it.